hepcrawl/spiders/base_spider.py

```diff
@@ -0,0 +1,229 @@
+                    surname, given_names = split_fullname(
+                        cleaned_author)
+                    authors.append({
+                        'surname': surname,
+                        'given_names': given_names,
+                        'full_name': cleaned_author,
+                    })
+        return authors
+
+    def get_urls_in_record(self, node):
+        """Return all the different urls in the xml.
+
+        Urls might be stored in identifier, relation, or link element. Beware
+        the strange "filename.jpg.pdf" urls.
+        """
+        identifiers = [
+            identifier for identifier in node.xpath(".//dc:identifier/text()").extract()
+            if "http" in identifier.lower() and "front" not in identifier.lower() and
+            "jpg" not in identifier.lower()
+        ]
+        relations = [
+            s for s in " ".join(node.xpath(".//dc:relation/text()").extract()).split() if "http" in s and
+            "jpg" not in s.lower()
+        ]
+        links = node.xpath(".//base_dc:link/text()").extract()
+        urls_in_record = []
+        for url in identifiers + relations + links:
+            if url.startswith('<'):
+                url = url[1:]
+            if url.endswith('>'):
+                url = url[:-1]
+            if not url.startswith("http://") and not url.startswith("https://"):
+                url = "http://{0}".format(url)
+            if url not in urls_in_record:
+                urls_in_record.append(url)
+        return urls_in_record
+
+    def find_direct_links(self, urls_in_record):
+        """Determine if the XML file has a direct link."""
+        direct_link = []
+        for link in urls_in_record:
+            if "pdf" in get_mime_type(link) and "jpg" not in link.lower():
+                direct_link.append(link)
+        if direct_link:
+            self.logger.info("Found direct link(s): %s", direct_link)
+        else:
+            self.logger.info("Didn't find direct link to PDF")
+
+        return direct_link
+
+    def parse_node(self, response, node):
+        """Iterate through all the record nodes in the XML.
+
+        With each node it checks if direct link exists, and sends
+        a request to appropriate function to parse the XML.
+        """
+        urls_in_record = self.get_urls_in_record(node)
+        direct_link = self.find_direct_links(urls_in_record)
+
+        if not direct_link and urls_in_record:
+            # Probably all links lead to same place, so take first
+            link = urls_in_record[0]
+            request = Request(link, callback=self.scrape_for_pdf)
+            request.meta["urls"] = urls_in_record
+            request.meta["node"] = node
+            return request
+        else:
+            response.meta["direct_link"] = direct_link
+            response.meta["urls"] = urls_in_record
+            response.meta["node"] = node
+            return self.build_item(response)
+
+    def build_item(self, response):
```

Comment:
Ideally we will add the designated collection tags on INSPIRE in the `collections` field. See e.g. https://github.com/inspirehep/hepcrawl/blob/master/hepcrawl/spiders/wsp_spider.py#L174-L185

If you are unsure about which collections we have, see with @annetteholtkamp  (Usually we mark BOOK, THESIS, CORE, Published, ConferencePaper etc.)